### PR TITLE
feat: add arm64 build

### LINKED ISSUE #931: Multi arch image for pod identity components
**Describe the request**
Investigate and add support for multi-arch (amd64, arm64) pod identity component images.

**Explain why AAD Pod Identity needs it**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by aramase**
**Reason for Change**:

**Requirements**

- [x] squashed commits
- [x] included documentation
- [x] added unit tests and e2e tests (if applicable). See [test standard](https://github.com/Azure/aad-pod-identity/blob/master/CONTRIBUTING.md#test-standard) for more details.
- [x] ran `make precommit`

**Issue Fixed**:

fixes #931 

**Please answer the following questions with yes/no**:

Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

- [ ] yes
- [x] no

**Notes for Reviewers**:

### CODE REVIEW

**Comment by aramase** [comment]
```json
➜ docker manifest inspect aramase/mic:arch03
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 738,
         "digest": "sha256:a4bdd3e8e97d4a3fc9b8ed8bb3d2ff0b2cda534f8635e1cde823c3bad60a76d6",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:733f288f543716d0f665028ca24d022e2a8c2cb1d6b41dd4d950ce953317ce56",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 737,
         "digest": "sha256:eee4cca84ed8b794f1e1165be28ef639f8de141de2c32666b6806653946b71c9",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      }
   ]
}
```

File: Makefile [hunk 1]
```diff
@@ -122,33 +126,48 @@ precommit: build unit-test lint
 deepcopy-gen:
 	deepcopy-gen -i ./pkg/apis/aadpodidentity/v1/ -o . -O aadpodidentity_deepcopy_generated -p aadpodidentity
 
+.PHONY: docker-buildx-builder
+docker-buildx-builder:
+	docker run --rm --privileged docker/binfmt:820fdd95a9972a5308930a2bdfb8573dd4447ad3
+	if ! DOCKER_CLI_EXPERIMENTAL=enabled docker buildx ls | grep -q container-builder; then\
+		DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name container-builder --use;\
+	fi
```
Comment (chewong) [inline Makefile:134]: `DOCKER_CLI_EXPERIMENTAL` is already exported above

```suggestion
	if ! docker buildx ls | grep -q container-builder; then \
		docker buildx create --name container-builder --use; \
	fi
```

File: Makefile [hunk 2]
```diff
@@ -48,7 +48,11 @@ TOOLS_DIR := $(abspath ./.tools)
 
 # docker env var
 DOCKER_BUILDKIT = 1
-export DOCKER_BUILDKIT
+DOCKER_CLI_EXPERIMENTAL=enabled
```
Comment (chewong) [inline Makefile:51]: nit:
```suggestion
DOCKER_CLI_EXPERIMENTAL = enabled
```

File: Dockerfile [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
+
 WORKDIR /go/src/github.com/Azure/aad-pod-identity
 ADD . .
 RUN go mod download
 ARG IMAGE_VERSION
-RUN make build
+RUN apk add make && \
+    export GOOS=$TARGETOS && \
+    export GOARCH=$TARGETARCH && \
+    GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3); export GOARM=${GOARM:1} && \
```
Comment (chewong) [inline Dockerfile:18]: GOARM only accepts 5, 6, or 7 according to https://github.com/golang/go/wiki/GoArm

```suggestion
     export GOARM=$(echo ${TARGETPLATFORM} | cut -d / -f3 | tr -d 'v') && \
```

**Review by chewong — CHANGES_REQUESTED** [review]

**Review by chewong — APPROVED** [review]

**Pull request merged**